python/src/spark_rapids_ml/core.py

```diff
@@ -552,11 +663,15 @@ def _call_cuml_fit_func(
         array_order = self._fit_array_order()
 
         cuml_verbose = self.cuml_params.get("verbose", False)
+        use_sparse_array = (
+            alias.featureVectorType in dataset.schema.fieldNames()
```

Comment:
Does this give a false positive if there is a column with the name alias.featureVectorType?   Maybe check specifically that the feature column is of struct type.

Reply:
It seems fine because core.py converts the feature column and label column into alias columns, through dataset = dataset.select(*select_cols). The select_cols all use alias names. 

Revised the code to also check the existence of alias.featureVectorSize and alias.featureVectorIndices, given the VectorUDT features column will be unwrapped into four alias columns (i.e. featureVectorType, featureVectorSize, featureVectorIndices, and data).